### PR TITLE
Drop querystring in favor of URLSearch params

### PR DESCRIPTION
**Pull request by H-Shay**
Per the discussion at https://github.com/matrix-org/conference-bot/pull/202#discussion_r1407046517, drop `querystring` in favor of using `URLSearchParams`.

### CODE REVIEW

File: web/widgets.ts [hunk 1]
```diff
@@ -14,11 +14,19 @@ See the License for the specific language governing permissions and
...
 export const isWidget = widgetId && widgetId !== "$matrix_widget_id";
 
-const widgetQuery = qs.parse(window.location.hash.substring(1));
-export const addlQuery = Object.assign({}, qs.parse(window.location.search.substring(1)), widgetQuery);
+const widgetparams = new URLSearchParams(window.location.hash.substring(1))
+const widgetQuery = {}
+for (const entry of widgetparams.entries()) {
+    widgetQuery[entry[0]] = entry[1]
+}
+
```
Comment (H-Shay) [inline web/widgets.ts:26]: To the reviewer: be convinced that these are equivalent - I'm new to all libraries involved but based on my reading of the docs this should provide the same result as `qs.parse()` - but correct me if I am wrong!

File: web/widgets.ts [hunk 2]
```diff
@@ -14,11 +14,19 @@ See the License for the specific language governing permissions and
...
 export const widgetId = urlParams.get("widgetId");
 export const isWidget = widgetId && widgetId !== "$matrix_widget_id";
 
-const widgetQuery = qs.parse(window.location.hash.substring(1));
-export const addlQuery = Object.assign({}, qs.parse(window.location.search.substring(1)), widgetQuery);
+const widgetparams = new URLSearchParams(window.location.hash.substring(1))
+const widgetQuery = {}
+for (const entry of widgetparams.entries()) {
+    widgetQuery[entry[0]] = entry[1]
+}
```
Comment (Half-Shot) [inline web/widgets.ts:25]:
```suggestion
const widgetQuery = Object.fromEntries(widgetparams.entries());
```

is equivalent :)

File: web/widgets.ts [hunk 3]
```diff
@@ -14,11 +14,19 @@ See the License for the specific language governing permissions and
...
+
+const searchParams = new URLSearchParams(window.location.search.substring(1))
+const parsedSearch = {}
+for (const entry of searchParams.entries()) {
+    parsedSearch[entry[0]] = entry[1]
+}
+export const addlQuery = Object.assign({}, parsedSearch, widgetQuery);
```
Comment (Half-Shot) [inline web/widgets.ts:32]:
```suggestion
export const addlQuery = { ...parsedSearch, ...widgetQuery };
```

File: web/widgets.ts [hunk 4]
```diff
@@ -14,11 +14,19 @@ See the License for the specific language governing permissions and
...
+const widgetQuery = {}
+for (const entry of widgetparams.entries()) {
+    widgetQuery[entry[0]] = entry[1]
+}
+
+const searchParams = new URLSearchParams(window.location.search.substring(1))
+const parsedSearch = {}
+for (const entry of searchParams.entries()) {
+    parsedSearch[entry[0]] = entry[1]
+}
```
Comment (Half-Shot) [inline web/widgets.ts:31]:
```suggestion
const parsedSearch = Object.fromEntries(searchParams.entries());
```

**Review by Half-Shot — APPROVED** [review]
Couple of suggestions for cleaner code, otherwise great stuff :)

**Pull request merged**